content/custom-base-images.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Custom base images
+order: 17
+description: Learn how to customize your app's environment
+---
+
+Most Galaxy users run their app with the [default base image](/base-image-packages.html).  This provides a basic environment for running most Meteor apps.
+
+For full control over your app's environment, you can provide a custom base image instead.  This allows you to precisely control the set of system packages available to your app, customize exactly what happens when Galaxy builds your app's image at deploy time and when Galaxy runs your app, and even allows you to run non-Meteor software on Galaxy.
+
+A base image is a [Docker](https://www.docker.com/) image.  Base images must live on [Docker Hub](https://hub.docker.com/) and be publicly readable.  (This means they should not contain any secrets.)
+
+You build your base image using a Dockerfile.  You should use the Dockerfile to choose an operating system for your base image (such as `ubuntu:16.04`) and install packages.  You can look at the [source of the Galaxy default base image](https://github.com/meteor/galaxy-images) for inspiration.  The Dockerfile should also add a file called `/app/setup.sh` and specify a command to run with the `CMD` or `ENTRYPOINT` directives.  For more details, see the [Docker documentation](https://docs.docker.com/).
+
+Once you've built your base image and pushed it to Docker Hub, specify it in your settings.json file under `baseImage` inside `galaxy.meteor.com` (next to `env`).  You need to specify both the Docker repository (which generally includes your username) and the tag you want to use.  For example, your settings file may look like:
+
+```
+{
+  "galaxy.meteor.com": {
+    "env": { "MONGO_URL": "mongodb://..." },
+    "baseImage": {
+      "repository": "username/my-galaxy-base-image",
+      "tag": "v23"
+    }
+  }
+}
+```
+
+Docker allows you to push a new image to an existing tag. Currently, if you do this, Galaxy will not rebuild your app until you perform another `meteor deploy`.  We may change this policy in the future.  We recommend that you use a new tag each time your change your base image, and specify that tag rather than a mutable tag like `latest` in your `baseImage` specification.
+
+<h2 id="build">Build time behavior</h2>
+
+When you run `meteor deploy`, your local `meteor` command-line tool builds your app, bundles it into a [tarball](https://en.wikipedia.org/wiki/Tar_(computing%29), and uploads it to Galaxy.
+
+Galaxy then builds a Docker image for your app based on the default base image by running the `/app/setup.sh` script inside the image using `/bin/bash`. Every custom base image must contain `/bin/bash` and `/app/setup.sh`.  (We may relax this requirement in the future to allow `/app/setup.sh` to work with any shell, so you must make sure that `/app/setup.sh` has the executable bit set and that it has a proper [shebang line](https://en.wikipedia.org/wiki/Shebang_(Unix%29).)
```

Comment:
same tarball link not showing properly problem in previous line

in this line - shebang line missing a period (you can see how the period was nested in the wrong set of parenthesis above - should be outside the parenthesis)

Reply:
Link looks good to me.

The full sentence is in parentheses so I believe the period should be inside.